storagenode/monitor/monitor.go

```diff
@@ -107,6 +112,16 @@ func (service *Service) Run(ctx context.Context) (err error) {
 		service.log.Warn("Disk space is less than requested. Allocating space", zap.Int64("bytes", service.allocatedDiskSpace))
 	}
 
+	// Ensure the disk is at least 500GB in size, which is our current minimum required to be an operator
+	if service.allocatedDiskSpace < minimumDiskSpace {
+		service.log.Fatal("Total disk space less than required minimum", zap.Int64("bytes", minimumDiskSpace))
```

Comment:
don't use fatal. we should return an error

Reply:
Does that stop the node directly?

Reply:
i don't know, but functions should not as a side effect call os.exit. if a service dies, we should have the service running in the peer know that certain errors are fatal errors. but it should be up to the peer code about whether to exit. having this run method decide the process should end is overstepping it's contractual bounds in my opinion

Reply:
so i think if you want the node to die you should check out the peer code in the storage node and make sure that if the monitor Run method errors that it kills the storage node

Reply:
maybe the error group the peer has should have two types of errors, non-fatal and fatal? and this is a fatal error.